peer.go

```diff
@@ -221,23 +227,29 @@ func (l *PeerList) exists(hostPort string) (*peerScore, uint64, bool) {
 	return ps, score, ok
 }
 
-// updatePeer is called when there is a change that may cause the peer's score to change.
+// onPeerChange is called when there is a change that may cause the peer's score to change.
 // The new score is calculated, and the peer heap is updated with the new score if the score changes.
-func (l *PeerList) updatePeer(p *Peer) {
+func (l *PeerList) onPeerChange(p *Peer) {
 	ps, psScore, ok := l.exists(p.hostPort)
 	if !ok {
 		return
 	}
 
-	newScore := l.scoreCalculator.GetScore(p)
```

Comment:
@prashantv doing a double calculation nukes `TestPeerSelection()` ...:

```go
assert.EqualValues(t, 5, count.Load(),
	"Expect 5 exchange updates: peer add, init mex, new conn, ping, pong")
```

I don't want to change that to "9", which is what I got...